### PR TITLE
scaffold header and index

### PR DESCRIPTION
**Pull request by varffs**
- bug with the_time
- date needs localization
- needs custom tax for post source
- needs gallery feature
- header menu is only for posts

### CODE REVIEW

File: css/site.styl [hunk 1]
```diff
@@ -31,49 +31,111 @@ normalize()
...
 .col
-  float: left
-  margin-left: ($gutter/2)
-  margin-right: ($gutter/2)
-  min-height: 1px
+  flex: 1 1 auto
+  box-sizing: border-box
+  max-width: 100%
+  padding: 0 ($gutter / 2)
+
```
Comment (mikeyrayvon) [inline css/site.styl:116]: `.col` needs `margin-bottom: $gutter`

Comment (varffs) [inline css/site.styl:116]: No I don't like the pattern still and I already needed to remove it in this feature branch for it to look as it should. This isn't mobile first tho so maybe works better like that

File: css/site.styl [hunk 2]
```diff
@@ -31,49 +31,111 @@ normalize()
...
+.col-no-gutter
+  padding-left: 0
+  padding-right: 0
+
+.col-no-margin-bottom
+  margin-bottom: 0
+
```
Comment (mikeyrayvon) [inline css/site.styl]: ok then whats this class 4

Comment (varffs) [inline css/site.styl]: dunno not needed ;)

File: header.php [hunk 3]
```diff
@@ -56,6 +56,21 @@
...
+    <div class="row">
+      <div class="col col-s-3 col-m-2">
+        <h1 class="font-uppercase"><a href="<?php echo home_url(); ?>"><?php bloginfo('name'); ?></a></h1>
+      </div>
+      <div class="col col-s-6 col-m-7">
+        <ul id="header-menu" class="font-sans u-inline-list">
+          <li><a href="<?php echo site_url(); ?>">Home</a></li>
+          <li><a href="<?php echo site_url('/prensa'); ?>">Prensa</a></li>
+          <li><a href="<?php echo site_url('/contacto'); ?>">Contacto</a></li>
+          <li><a href="<?php echo site_url('/noticias'); ?>">Noticias</a></li>
```
Comment (mikeyrayvon) [inline header.php:69]: you dont like my pattern for this?

**Comment by mikeyrayvon** [comment]
if were going to use flex we should take advantage of its power and not just set col-num for everything. 

i think we take the `col-size-num` classes off these and set `flex-shrink: 1; flex-grow: 0` for the outer cols, and `flex-shrink: 0; flex-grow: 1` for the middle one

cause its line-breaking on my screen
screen shot 2016-08-17 at 1 45 06 pm

**Comment by mikeyrayvon** [comment]
maybe we need `col-s-auto` `col-m-auto` `col-l-auto` classes to reset the `flex-basis` by breakpoint?

**Comment by mikeyrayvon** [comment]
and then `col-s-shrink-1`, `col-s-shrink-0`, `col-s-grow-0`, `col-s-grow-1` etc?

**Comment by mikeyrayvon** [comment]
and `.col` should probably be `flex: 0 1 auto`

File: header.php (hunk 3 above)
Comment (varffs) [inline header.php:69]: what pattern?

**Comment by varffs** [comment]
I copied the col declaration from one of the 2 flex projects as not in boil yet

**Comment by mikeyrayvon** [comment]
i know but i think we should rework this

**Comment by mikeyrayvon** [comment]
do you mind if i do like i just said?

**Comment by varffs** [comment]
ref flexing auto I dunno. I like columnized design so don't really want it resizing all on its own. If the did that in the header then the main content grid won't align which is not what we want

**Comment by mikeyrayvon** [comment]
right

**Comment by mikeyrayvon** [comment]
i think we should have the option in the boiler plate. cause whats the point of flex if it cant flex ever?

File: header.php (hunk 3 above)
Comment (mikeyrayvon) [inline header.php:69]: using `get_id_by_slug('page_slug')` to check if the page exists, then `echo get_permalink(get_id_by_slug('page_slug')` if it exists

Comment (mikeyrayvon) [inline header.php:69]: so we dont have broken links in the header

**Comment by varffs** [comment]
k I think this is there now. 

I like the fixed breakpoints all the way for each of the most common screen sizes. 

And I've tried to put the mobile in another responsive/x.styl so universal classes can go in the main.styl and all layout and margin/padding can do in responsive/x.styl files.

**Pull request merged**